cross_compile/sysroot_compiler.py

```diff
@@ -102,21 +109,51 @@ class DockerConfig:
     3. Setting to enable/disable caching during docker build
     """
 
-    _default_docker_base_image = {
-        ('armhf', 'ubuntu'): 'arm32v7/ubuntu:bionic',
-        ('armhf', 'debian'): 'arm32v7/debian:latest',
-        ('aarch64', 'ubuntu'): 'arm64v8/ubuntu:bionic',
-        ('aarch64', 'debian'): 'arm64v8/debian:latest',
-    }  # type: Dict[tuple, str]
+    @staticmethod
+    def _get_docker_base_image(arch, os, rosdistro) -> str:
+        """
+        Return docker image to use in dockerfile used to cross compile.
+
+        Args:
+            arch: Target architecture of cross compiling
+            os: Target OS of cross compiling
+            rosdistro: ROS distribution that is being cross compiled
+
+        Returns:
+            image_name: Docker image name to use in the dockerfile
+        """
+        base_image = ''
+        tag = ''
+
+        if str(arch) == 'armhf':
```

Comment:
i think this should be structured as a lookup table instead of if..elif blocks, e.g.

```
base_repo = {
  'armhf': 'arm32v7',
  'aarch64': 'arm64v8',
}
image_tag = {
  'kinetic': {
    'ubuntu': 'xenial',
    'debian': 'jessie',
  },
  'melodic': {
    'ubuntu': 'bionic',
    'debian': 'stretch',
  },
  etc...
}

image = "{}/{}:{}".format(base_repo[arch], os, image_tag[rosdistro][os])
```

better to keep configuration as data rather than as code